src/seccomp/src/bin.rs

```diff
@@ -0,0 +1,567 @@
+        input_file: input_file.unwrap(),
+        // Safe to unwrap because it has a default value
+        output_file: arguments.value_as_string("output_file").unwrap(),
+    })
+}
+
+fn parse_json(reader: &mut dyn Read) -> Result<HashMap<String, Filter>> {
+    serde_json::from_reader(reader).map_err(Error::JSON)
+}
+
+fn compile(args: &Arguments) -> Result<()> {
+    let input_file = File::open(&args.input_file)
+        .map_err(|err| Error::FileOpen(PathBuf::from(&args.input_file), err))?;
+    let mut input_reader = BufReader::new(input_file);
+    let filters = parse_json(&mut input_reader)?;
+    let compiler = Compiler::new(&args.target_arch);
+
+    // transform the IR into a Map of BPFPrograms
+    let bpf_data: BpfThreadMap = compiler.compile_blob(filters).map_err(Error::FileFormat)?;
+
+    // serialize the BPF programs & output them to a file
+    let output_file = File::create(&args.output_file)
+        .map_err(|err| Error::FileOpen(PathBuf::from(&args.output_file), err))?;
+    bincode::serialize_into(output_file, &bpf_data).map_err(Error::Bincode)?;
+
+    Ok(())
+}
+
+fn main() {
+    let mut arg_parser = build_arg_parser();
+
+    if let Err(err) = arg_parser.parse_from_cmdline() {
+        println!(
+            "Arguments parsing error: {} \n\n\
+             For more information try --help.",
+            err
+        );
+        process::exit(1);
+    }
+
+    if let Some(help) = arg_parser.arguments().value_as_bool("help") {
+        if help {
+            println!("Seccompiler v{}\n", SECCOMPILER_VERSION);
+            println!("{}", arg_parser.formatted_help());
+            process::exit(0);
+        }
+    }
+    if let Some(version) = arg_parser.arguments().value_as_bool("version") {
+        if version {
+            println!("Seccompiler v{}\n", SECCOMPILER_VERSION);
+            process::exit(0);
+        }
+    }
+
+    let args = get_argument_values(arg_parser.arguments()).unwrap_or_else(|err| {
+        println!(
+            "{} \n\n\
+            For more information try --help.",
+            err
+        );
+        process::exit(1);
+    });
+
+    compile(&args).unwrap_or_else(|err| panic!("Seccompiler error: {}", err));
+
+    println!("Filter successfully compiled into: {}", args.output_file);
+}
+
+#[cfg(test)]
+mod tests {
+    use super::compiler::{Compiler, Error as FilterFormatError, Filter, SyscallObject};
+    use super::BpfThreadMap;
+    use super::{build_arg_parser, compile, get_argument_values, parse_json, Arguments, Error};
+    use bincode::Error as BincodeError;
+    use seccomp::{
+        deserialize_binary, SeccompAction, SeccompCmpArgLen::*, SeccompCmpOp::*,
+        SeccompCondition as Cond,
+    };
+    use std::collections::HashMap;
+    use std::io;
+    use std::io::Write;
+    use std::path::PathBuf;
+    use utils::tempfile::TempFile;
+
+    // test helper for generating correct JSON input data
+    fn get_correct_json_input() -> String {
```

Comment:
```suggestion
    fn default_json_input() -> String {
```

Reply:
hmm, this input is in no way default. It's just an example of a correct filter file